core/src/main/java/com/graphhopper/reader/OSMTurnRelation.java

```diff
@@ -0,0 +1,151 @@
+        this.fromOsmWayId = fromWayID;
+        this.viaOsmNodeId = viaNodeID;
+        this.toOsmWayId = toWayID;
+        this.restriction = restrictionType;
+        this.vehicleTypeRestricted = vehicleTypeRestricted;
+        this.vehicleTypesExcept = vehicleTypesExcept;
+    }
+
+    public long getOsmIdFrom() {
+        return fromOsmWayId;
+    }
+
+    public long getOsmIdTo() {
+        return toOsmWayId;
+    }
+
+    public long getViaOsmNodeId() {
+        return viaOsmNodeId;
+    }
+
+    public Type getRestriction() {
+        return restriction;
+    }
+
+    public String getVehicleTypeRestricted() {
+        return vehicleTypeRestricted;
+    }
+
+    public List<String> getVehicleTypesExcept() {
+        return vehicleTypesExcept;
+    }
+
+    /**
+     * For a conditional turn restriction, test each vehicle type to verify if it is concerned.
+     * For a normal turn restriction (non conditional), the restriction is necessary considered.
+     * @param vehicleTypes
+     * @return
+     */
+    public boolean isVehicleTypeConcernedByTurnRestriction(List<String> vehicleTypes) {
+        if (Helper.isEmpty(vehicleTypeRestricted) && ((vehicleTypesExcept == null) || ((vehicleTypesExcept != null) && (vehicleTypesExcept.isEmpty())))) {
+            return true;
+        }
+        for (String vehicleType : vehicleTypes) {
+            if (!Helper.isEmpty(vehicleTypeRestricted)) {
+                if (vehicleTypeRestricted.equals(vehicleType)) {
+                    return true;
+                }
+            }
+            if (vehicleTypesExcept != null) {
+                if (vehicleTypesExcept.contains(vehicleType)) {
+                    return false;
+                }
+            }
+        }
+        if (!Helper.isEmpty(vehicleTypeRestricted)) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return "*-(" + fromOsmWayId + ")->" + viaOsmNodeId + "-(" + toOsmWayId + ")->*";
+    }
+
+    public enum Type {
+        UNSUPPORTED, NOT, ONLY;
+
+        private static final Map<String, Type> tags = new HashMap<>();
+
+        static {
+            tags.put("no_left_turn", NOT);
+            tags.put("no_right_turn", NOT);
+            tags.put("no_straight_on", NOT);
+            tags.put("no_u_turn", NOT);
+            tags.put("only_right_turn", ONLY);
+            tags.put("only_left_turn", ONLY);
+            tags.put("only_straight_on", ONLY);
+        }
+
+        public static Type getRestrictionType(String tag) {
+            Type result = null;
+            if (tag != null) {
+                result = tags.get(tag);
+            }
+            return (result != null) ? result : UNSUPPORTED;
+        }
+    }
+
+    /**
+     * Helper class to processing purposes only
+     */
+    public static class TurnCostTableEntry {
```

Comment:
This can probably be moved into OSMReader now ?

Reply:
ok